### DISCUSSION TITLE
Determining correct order of Trilinos libraries in link line

### DISCUSSION THREAD

**Discussion by jhux2** [trilinos/Trilinos#13054] [Q&A] [answered]
@trilinos/framework @bartlettroscoe @abodado

Is there a mechanism for an application building against an installed Trilinos build to identify the proper link ordering for Trilinos libraries?  If it matters, the customer is doing a static build.  I don't know if the app itself is using cmake.

**Comment by bartlettroscoe** [comment, marked as answer]
Currently the only supported approach is to have downstream apps use CMake and pull in Trilinos pacakges with `find_package(<PackageName>)` and then link to `target_link_libraries(<yourLibOrExec> PRIVATE <PackageName>::all_libs ...)`.  This was one of the consequences of moving to modern CMake.  See the discussion of this in https://github.com/TriBITSPub/TriBITS/issues/299 starting around https://github.com/TriBITSPub/TriBITS/issues/299#issuecomment-582558675 (and see Kitware's view on this in https://github.com/TriBITSPub/TriBITS/issues/299#issuecomment-582577313).
